### PR TITLE
[SuperTextField][iOS] Fix toolbar focal point for collapsed selections (Resolves #1683)

### LINKED ISSUE #1683: [SuperTextField] [iOS] Toolbar looks broken when opened with collapsed selection
**Package Version**
Using 0a58e704469ccaffe5ab14620a56afe75bee6574 in `main`.

**To Reproduce**
1. Go to `SuperTextField` demo for iOS
2. Place the selection in the text
3. Wait enough time to not trigger double-tap, and tap once on the the caret

**Actual behavior**
See screenshot below.

**Expected behavior**
Shows the toolbar as expected.

**Platform**
Which device(s) and operating system(s) have experienced the problem?
iOS. Did not test on Android, so I'm not sure if there's an issue there.

**Flutter version**
Flutter `master` commit e278279a48f0479386c249cf18f26b29c6f9018f.

**Screenshots**
simulator_screenshot_B0888A53-2A79-4DFE-8917-DFB381BF23DF

### PR DESCRIPTION
**Pull request by angelosilvestre**
[SuperTextField][iOS] Fix toolbar focal point for collapsed selections. Resolves #1683

The iOS floating toolbar is broken for collapsed selections:

simulator_screenshot_B0888A53-2A79-4DFE-8917-DFB381BF23DF

The cause is that we aren't computing the selection rect for collapsed selections.

This PR changes the iOS textfield to place the selection rect at the caret top for collapsed selections.

### CODE REVIEW

File: super_editor/test_goldens/super_textfield/super_textfield_toolbar_test.dart [hunk 1]
```diff
@@ -29,7 +30,34 @@ void main() {
       );
```
Comment (matthew-carroll) [inline super_editor/test_goldens/super_textfield/super_textfield_toolbar_test.dart:30]: Should the test file above this include a reference to an "expanded" selection?

Comment (angelosilvestre) [reply super_editor/test_goldens/super_textfield/super_textfield_toolbar_test.dart:30]: I think I didn't understand what you mean.

Comment (matthew-carroll) [reply super_editor/test_goldens/super_textfield/super_textfield_toolbar_test.dart:30]: Check the golden file name immediately above this comment. Should that golden file name include the word "expanded" somewhere in it?

Comment (angelosilvestre) [reply super_editor/test_goldens/super_textfield/super_textfield_toolbar_test.dart:30]: Updated.

File: super_editor/lib/src/super_textfield/ios/_user_interaction.dart [hunk 2]
```diff
@@ -477,11 +477,21 @@ class IOSTextFieldTouchInteractorState extends State<IOSTextFieldTouchInteractor
...
+    if (!widget.textController.selection.isValid) {
       _toolbarFocusSelectionRect.value = null;
       return;
     }
 
+    if (widget.textController.selection.isCollapsed) {
+      // The selection is collapsed.
+      // Place the selection rect at the top of the caret position.
+      final caretOffset = _textLayout.getOffsetForCaret(widget.textController.selection.extent);
+      _toolbarFocusSelectionRect.value = caretOffset & Size.zero;
```
[the remark this answers is not in the document]
Comment (angelosilvestre) [reply super_editor/lib/src/super_textfield/ios/_user_interaction.dart]: Updated to do the same as `SuperEditor` does.

**Review by matthew-carroll — APPROVED** [review]
LGTM

**Pull request merged**